src/vehicle_move.cpp

```diff
@@ -512,23 +512,28 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
         ret.target = critter;
         e = 0.30;
         part_dens = 15;
-        switch( critter->get_size() ) {
-            case MS_TINY:    // Rodent
-                mass2 = 1;
-                break;
-            case MS_SMALL:   // Half human
-                mass2 = 41;
-                break;
-            default:
-            case MS_MEDIUM:  // Human
-                mass2 = 82;
-                break;
-            case MS_LARGE:   // Cow
-                mass2 = 400;
-                break;
-            case MS_HUGE:     // TAAAANK
-                mass2 = 1000;
-                break;
+        monster *mon = dynamic_cast<monster *>( critter );
+        if( mon ) {
```

Comment:
if a Creature is at this location, there are 3 choices for what it could be: npc, monster, or avatar. in all three situations, the object has the get_weight() function. therefore, the switch statement should go away and error if somehow the creature isn't any of those three.